Classes/Private/Manager/Objects/OCTActiveFile.m

```diff
@@ -0,0 +1,530 @@
+//
+//  OCTActiveFile.m
+//  objcTox
+//
+//  Created by stal on 9/7/2015.
+//  Copyright © 2015 Zodiac Labs. All rights reserved.
+//
+
+#import "DDLog.h"
+#import "OCTTox.h"
+#import "OCTActiveFile+Variants.h"
+#import "OCTSubmanagerFiles.h"
+#import "OCTMessageFile.h"
+#import "OCTSubmanagerFiles+Private.h"
+#import "OCTSubmanagerObjects+Private.h"
+#import "OCTRealmManager.h"
+
+#include <mach/clock.h>
+#include <mach/mach_host.h>
+#include <mach/mach_port.h>
+
+#undef LOG_LEVEL_DEF
+#define LOG_LEVEL_DEF LOG_LEVEL_DEBUG
+
+/* these are annoying, and because we get all file numbers from toxcore,
+ * casting will never truncate them because they were never longs to begin
+ * with. */
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wshorten-64-to-32"
+
+static time_t OCTGetSystemUptime(void)
```

Comment:
To properly count progress, a clock source that doesn't go back in time (the user can change the system date) is needed. The Mach clock is guaranteed to not do that.

(That said, the function name is wrong so I'll need to fix that)